libs/x-content/src/main/java/org/elasticsearch/xcontent/support/filtering/FilterPath.java

```diff
@@ -168,18 +168,14 @@ static void insertNode(String filter, BuildNode node) {
 
             if (splitPosition > 0) {
                 String field = findEscapes
```

Comment:
The formatter wants to put the ternary `x ? y : z` expression on one line, now that it's shorter. You can run `./gradlew spotlessApply` to make this change automatically.

Reply:
I am sorry. I will fix it in a moment.
(Still trying to figure out how to make IntelliJ IDEA run this 'spotlessApply' before commit.)

Reply:
This potentially adds some overhead to gradle runs, but I've put the following in `~/.gradle/init.d/spotless.gradle` and it has definitely reduced the number of formatting-only commits I've had to make:

```
allprojects {
  // run spotlessApply before all java compilation
  pluginManager.withPlugin('com.diffplug.spotless') {
    tasks.withType(JavaCompile).configureEach {
      dependsOn('spotlessApply')
    }
  }
}
```

I know some other team members prefer to put a spotless check in their pre-push git commit hook.

Reply:
Thank you for the tip! I totally missed the pre-comit hook option. Probably because I do not use Git hooks that much.

I am currently checking IDEA's "Reformat code" commit check. I was afraid it would use built-in code formatter.
But after reading the Eclipse Code Formatter plugin docs it seems the `formatterConfig.xml` should be used.